src/root/rmem.d

```diff
@@ -127,7 +127,7 @@ else
     __gshared size_t heapleft = 0;
     __gshared void* heapp;
 
-    extern (C++) void* allocmemory(size_t m_size)
+    extern (C) void* allocmemory(size_t m_size)
```

Comment:
I meant `d_size_t`, not `dsize_t`.  I don't care about the name, at all.  What I find ridiculous is that you've introduced a new convention to use `d_size_t` in this PR but for some reason in this one place you've instead opted to change the linkage instead.  Be consistent.